spec/support/shared_examples_for_recovery_codes.rb

```diff
@@ -71,6 +71,18 @@
         expect(page).not_to have_xpath("//#{invisible_selector}[@id='recovery-code']")
       end
 
+      scenario 'focus is on first input and is trapped in modal' do
+        click_acknowledge_recovery_code
+
+        expect(page.evaluate_script('document.activeElement.name')).to eq 'recovery-0'
+
+        body_element = page.find('body')
+        body_element.send_keys [:shift, :tab]
+        expect(page.evaluate_script('document.activeElement.innerText')).to eq(
```

Comment:
would it make sense to also test that closing the modal does not "trap" the focus? I think that is what you are setting up when you have ` this.trap[showing ? 'activate' : 'deactivate']();` so would be good to have test coverage for that, too!